tests/acceptance/standard/test_hooks_processor.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from typing import TYPE_CHECKING
+
+from gitlab.v4.objects import ProjectHook
+
+from tests.acceptance import run_gitlabform, get_random_name
+
+
+class TestHooksProcessor:
+    def project_hook(self, project):
+        name = get_random_name("hook")
+        hook = project.hooks.create({"url": f"http://hooks/{name}.org"})
+
+        return hook
+
+    def test_project_hooks_create_delete(self, gl, project):
+        first_hook = self.project_hook(project)
+        second_hook = self.project_hook(project)
+
+        target = project.path_with_namespace
+        test_yaml = f"""
+            projects_and_groups:
+              {target}:
+                hooks:
+                  {first_hook.url}:
+                    push_events: {first_hook.push_events}
+                    merge_requests_events: {first_hook.merge_requests_events}
+                  {second_hook.url}:
+                    push_events: {second_hook.push_events}
+                    merge_requests_events: {second_hook.merge_requests_events}
+            """
+
+        run_gitlabform(test_yaml, target)
+
+        first_created_hook = project.hooks.get(first_hook.id)
+        second_created_hook = project.hooks.get(second_hook.id)
+
+        if TYPE_CHECKING:
+            assert isinstance(first_created_hook, ProjectHook)
+            assert isinstance(second_created_hook, ProjectHook)
+        assert len(project.hooks.list()) == 2
+        assert first_hook == first_created_hook
+        assert second_hook == second_created_hook
+
+        delete_yaml = f"""
+        projects_and_groups:
+            {target}:
+              hooks:
+                {first_hook.url}:
+                  delete: true
+        """
```

Comment:
I see... I thought this was exactly what I was trying to test, that even if the second hook was not included, it would not get deleted because it doesn't come with a delete: true in the config. I didn't know they could get deleted another way. OK, I'm happy to try to add the support, when this is done.